### PR TITLE
Add checkstyle validation and fix style violations in the pulsar client api module

### PR DESCRIPTION
**Pull request by vzhikserg**
### Modifications

The following changes were made in this pull request:

* Configured the checkstyle plugin
* Fixed reported style violations

This pull request has only cosmetic changes in the pulsar client api module and shouldn't introduce any change in the business logic.

### CODE REVIEW

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java [hunk 1]
```diff
@@ -61,6 +61,7 @@ public static Authentication token(Supplier<String> tokenSupplier) {
      *            the path to the TLS client private key
      * @return the Authentication object initialized with the TLS credentials
      */
+    @SuppressWarnings("checkstyle:MethodName")
```
Comment (sijie) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java]: Use `CHECKSTYLE.OFF` and `CHECKSTYLE.ON` comments.

```
// CHECKSTYLE.OFF: MethodName
...
// CHECKSTYLE.ON: MethodName
```

Comment (vzhikserg) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java]: Thank you for the recommendation! What is the best way to use these comments?
```
// CHECKSTYLE.OFF: MethodName
public static Authentication TLS(String certFilePath, String keyFilePath) {
// CHECKSTYLE.ON: MethodName
// ... the rest of the method
}
```
or
```
// CHECKSTYLE.OFF: MethodName
public static Authentication TLS(String certFilePath, String keyFilePath) {
// ... the rest of the method
}
// CHECKSTYLE.ON: MethodName
```

Comment (sijie) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java]: the later one?

Comment (vzhikserg) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java]: The code was updated

Comment (vzhikserg) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java]: The code was updated

**Comment by vzhikserg** [comment]
run integration tests

**Pull request merged**